### PR TITLE
docs: add warning for msal ClientId

### PR DESCRIPTION
**Pull request by rajamatt**
GitHub Issue (If applicable): #2356

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/Learn/Authentication/HowTo-MsalAuthentication.md [hunk 1]
```diff
@@ -88,6 +88,35 @@ uid: Uno.Extensions.Authentication.HowToMsalAuthentication
...
+                    );
+                });
+            });
+        ...
+    }
+    ```
+
+    > [!WARNING]
+    > A ClientId of GUID format is required for MSAL Authentication to work. You can specify it in the appsettings.json file, or in the code itself.
+    > If the ClientId cannot be found, the following warning will be logged: _Uno.Extensions.Authentication.MSAL.MsalAuthenticationProvider[0] A valid ClientId (GUID) is required for MSAL. Please provide the ClientId in the appsettings.json file._
```
Comment (carldebilly) [inline doc/Learn/Authentication/HowTo-MsalAuthentication.md:118]:
```suggestion
    > If the ClientId cannot be found, the following warning will be logged:
    > ```
    > _Uno.Extensions.Authentication.MSAL.MsalAuthenticationProvider[0] A valid ClientId (GUID) is required for MSAL. Please provide the ClientId in the appsettings.json file._
    > ```

**Review by carldebilly — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/4.1

File: doc/Learn/Authentication/HowTo-MsalAuthentication.md [hunk 2]
```diff
@@ -88,6 +88,39 @@ uid: Uno.Extensions.Authentication.HowToMsalAuthentication
...
+                    );
+                });
+            });
+        ...
+    }
+    ```
+
+    > [!WARNING]
+    > A ClientId of GUID format is required for MSAL Authentication to work. You can specify it in the appsettings.json file, or in the code itself.
+    > If the ClientId cannot be found, the following warning will be logged:
```
Comment (nickrandolph) [inline doc/Learn/Authentication/HowTo-MsalAuthentication.md:118]: This isn't correct - the application will crash with an exception. The details about the missing clientid are in the exception

Comment (rajamatt) [reply doc/Learn/Authentication/HowTo-MsalAuthentication.md]: I've updated it to be the error instead

I didn't include the part about not being able to check for the ClientId on our side but I can put it in there if we should mention it as well

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**